src/StdStorage.sol

```diff
@@ -68,16 +68,21 @@ library stdStorageSafe {
                 if (prev == bytes32(0)) {
                     emit WARNING_UninitedSlot(who, uint256(reads[i]));
                 }
+                if (prev != fdat) {
+                    continue;
+                }
```

Comment:
Why do we need to add this check? (I'm not too familiar with the internals of this portion)

Reply:
When finding the target slot, we should ensure the slot value == function return value, just as we did in the first if block. This could reduce some unnecessary SSTOREs and checks in the loop.

Another reason is that only checking the slot value after modifying it to a new value isn't enough. Consider this case:

```
contract Test {
    bytes public a  = hex"1";
    bytes public b = ~a;
}
```

If we're finding the slot ID of b and we're iterating through `[a, b]`, setting `a` to `~a` would result in the equation to slot value b (function return value). And `find()` would just return the slot ID of a instead of b. Require `prev == fdat` could help to correct it in this case.